File: sdks/php/src/Types/Response.php
<?php

declare(strict_types=1);

namespace Blok\Nanoservice\Types;

/**
 * Response represents the workflow response.
 */
final class Response
{
    public function __construct(
        public mixed $data = null,
        public string $contentType = '',
        public bool $success = false,
        public mixed $error = null,
    ) {}

    /**
     * Create a Response from an associative array.
     */
    public static function fromArray(array $data): self
    {
        return new self(
            data: $data['data'] ?? null,
            contentType: $data['contentType'] ?? '',
            success: $data['success'] ?? false,
            error: $data['error'] ?? null,
        );
    }

    /**
     * Convert to an associative array for JSON serialization.
     */
    public function toArray(): array
    {
        return [
            'data' => $this->data,
            'contentType' => $this->contentType,
            'success' => $this->success,
            'error' => $this->error,
        ];
    }
}

File: sdks/csharp/src/Nanoservice.Core/Types/Response.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Nanoservice.Core.Types;

/// <summary>
/// Response represents the workflow response.
/// </summary>
public class Response
{
    [JsonPropertyName("data")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public JsonElement? Data { get; set; }

    [JsonPropertyName("contentType")]
    public string ContentType { get; set; } = string.Empty;

    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("error")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public JsonElement? Error { get; set; }
}
